awx/ui/client/features/templates/list-templates.controller.js

```diff
@@ -0,0 +1,203 @@
+    };
+
+    // get pretified template type names from options
+    vm.templateTypes = unifiedJobTemplate.model.OPTIONS.actions.GET.type.choices
+        .reduce((acc, i) => {
+            acc[i[0]] = i[1];
+            return acc;
+        }, {});
+
+    // get if you should show the active indicator for the row or not
+    // TODO: edit indicator doesn't update when you enter edit route after initial load right now
+    vm.activeId = parseInt($state.params.job_template_id || $state.params.workflow_template_id);
+
+    // TODO: update to new way of launching job after mike opens his pr
+    vm.submitJob = function(template) {
+        if(template) {
+            if(template.type && (template.type === 'Job Template' || template.type === 'job_template')) {
+                InitiatePlaybookRun({ scope: $scope, id: template.id, job_type: 'job_template' });
+            }
+            else if(template.type && (template.type === 'Workflow Job Template' || template.type === 'workflow_job_template')) {
+                InitiatePlaybookRun({ scope: $scope, id: template.id, job_type: 'workflow_job_template' });
+            }
+            else {
+                // Something went wrong - Let the user know that we're unable to launch because we don't know
+                // what type of job template this is
+                Alert('Error: Unable to determine template type', 'We were unable to determine this template\'s type while launching.');
+            }
+        } else {
+            Alert('Error: Unable to launch template', 'Template parameter is missing');
+        }
+    };
+
+    // TODO: implement copy function
+    vm.copyTemplate = function(template) {
+    };
+
+    vm.deleteTemplate = function(template) {
+        var action = function() {
+            $('#prompt-modal').modal('hide');
+            Wait('start');
+            // TODO: The request url doesn't work here
+            unifiedJobTemplate.request('delete', template.id)
+                .then(() => {
+
+                    let reloadListStateParams = null;
+
+                    if($scope.templates.length === 1 && $state.params.template_search && !_.isEmpty($state.params.template_search.page) && $state.params.template_search.page !== '1') {
+                        reloadListStateParams = _.cloneDeep($state.params);
+                        reloadListStateParams.template_search.page = (parseInt(reloadListStateParams.template_search.page)-1).toString();
+                    }
+
+                    if (parseInt($state.params.template_id) === template.id) {
+                        $state.go("^", reloadListStateParams, { reload: true });
+                    } else {
+                        $state.go('.', reloadListStateParams, {reload: true});
+                    }
+                })
+                .catch(({data, status}) => {
+                    // TODO: Translate
+                    ProcessErrors($scope, data, status, null, {
+                        hdr: `Error!`,
```

Comment:
what would be the best way to do string translations here @gconsidine?

Reply:
If it's a general error message you could define it in the base string file: https://github.com/ansible/awx/blob/devel/awx/ui/client/lib/services/base-string.service.js#L57 -- otherwise, the templates strings file you created is fine.